src/Microsoft.AspNet.OData.Shared/EdmEntityObject.cs

```diff
@@ -42,6 +46,52 @@ public EdmEntityObject(IEdmEntityTypeReference edmType)
         public EdmEntityObject(IEdmEntityType edmType, bool isNullable)
             : base(edmType, isNullable)
         {
+            PersistentInstanceAnnotationsContainer = new ODataInstanceAnnotationContainer();
+            TransientInstanceAnnotationContainer = new ODataInstanceAnnotationContainer();
```

Comment:
I don't know how much memory a `new ODataInstanceAnnotationContainer` consumes but an alternative to doing this could be to delay the initialization until the respective properties are accessed, i.e.,
```csharp
private IODataInstanceAnnotationContainer persistentInstanceAnnotationsContainer;
private IODataInstanceAnnotationContainer transientInstanceAnnotationsContainer;

internal IODataInstanceAnnoationContainer PersistentInstanceAnnotationsContainer
{
    get
    {
        if (persistentInstanceAnnotationsContainer == null)
        {
            persistentInstanceAnnotationsContainer = new ODataInstanceAnnotationContainer();
        }

        return persistentInstanceAnnotationsContainer;
    }

    set { persistentInstanceAnnotationsContainer = value; }
}

internal IODataInstanceAnnoationContainer TransientInstanceAnnotationsContainer
{
    get
    {
        if (transientInstanceAnnotationsContainer == null)
        {
            transientInstanceAnnotationsContainer = new ODataInstanceAnnotationContainer();
        }

        return transientInstanceAnnotationsContainer;
    }

    set { transientInstanceAnnotationsContainer = value; }
}
```